File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/common/before_launch/RunMavenGoalSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.nio.file.Path;

@Value.Immutable
@JsonDeserialize(as = ImmutableRunMavenGoalSettings.class)
public interface RunMavenGoalSettings extends BeforeLaunchConfigurationSettings {

  Path workingDirectory();

  String commandLine();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/common/before_launch/RunMavenGoalSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch

import com.fasterxml.jackson.annotation.JsonProperty
import java.nio.file.Path

data class RunMavenGoalSettings(

    @JsonProperty("Working directory")
    val workingDirectory: Path,

    @JsonProperty("Command line")
    val commandLine: String
) : BeforeLaunchConfigurationSettings
